main.go

```diff
@@ -102,308 +54,94 @@ func exportEnvironmentWithEnvman(keyStr, valueStr string) error {
 	return cmd.Run()
 }
 
-// -----------------------
-// --- Main
-// -----------------------
-func main() {
-	//
-	// Validate options
-	configs := createConfigsModelFromEnvs()
-
-	fmt.Println()
-	configs.print()
-
-	if err := configs.validate(); err != nil {
-		log.Errorf("Issue with input: %s", err)
-		os.Exit(1)
+func mainE() error {
+	if err := initConfig(); err != nil {
+		return fmt.Errorf("Failed, error: %v", err)
 	}
-	fmt.Println()
-	// ---
-
-	// git
-	log.Infof("Git clone repository")
 
-	git, err := gitutil.NewHelper(configs.CloneIntoDir, configs.RepositoryURL, configs.ResetRepository == "Yes", configs.ManualMerge == "yes")
+	originPresent, err := isOriginPresent(config.CloneIntoDir, config.RepositoryURL)
 	if err != nil {
-		log.Errorf("Failed to create git helper, error: %s", err)
-		os.Exit(1)
+		return fmt.Errorf("Can't check if origin is presented, error: %v", err)
 	}
 
-	if err := git.Init(); err != nil {
-		log.Errorf("Failed, error: %s", err)
-		os.Exit(1)
-	}
-
-	git.ConfigureCheckout(configs.PullRequestID, configs.PullRequestURI, configs.PullRequestMergeBranch, configs.Commit, configs.Tag, configs.Branch, configs.BranchDest, configs.CloneDepth, configs.BuildURL, configs.BuildAPIToken)
-
-	if !git.IsOriginPresented() {
-		if err := git.RemoteAdd(); err != nil {
-			log.Errorf("Failed, error: %s", err)
-			os.Exit(1)
+	if originPresent && config.ResetRepository() {
+		if err := resetRepo(); err != nil {
+			return fmt.Errorf("Can't reset repository, error: %v", err)
 		}
 	}
 
-	if err := fetchWithRetry(git); err != nil {
-		if configs.PullRequestID != "" && configs.PullRequestMergeBranch != "" {
-			log.Warnf("Failed to fetch pull request, this happens most likely because the pull request is closed or has conflict.")
-		}
-		log.Errorf("Failed, error: %s", err)
-
-		if git.ManualMerge && configs.PullRequestID != "" && configs.PullRequestMergeBranch != "" {
-			log.Warnf("Failed to fetch repository (%s), set manual_merge to 'no'", configs.PullRequestURI)
-			fmt.Println()
-
-			git.ManualMerge = false
-			if err := git.RemoteRemove("origin"); err != nil {
-				log.Errorf("Failed, error: %s", err)
-				os.Exit(1)
-			}
-
-			git.ConfigureCheckout(configs.PullRequestID, configs.PullRequestURI, configs.PullRequestMergeBranch, configs.Commit, configs.Tag, configs.Branch, configs.BranchDest, configs.CloneDepth, configs.BuildURL, configs.BuildAPIToken)
-			git.SetRemoteURI(configs.RepositoryURL)
-			if err := git.RemoteAdd(); err != nil {
-				log.Errorf("Failed, error: %s", err)
-				os.Exit(1)
-			}
-			if err := fetchWithRetry(git); err != nil {
-				log.Errorf("Failed, error: %s", err)
-				os.Exit(1)
-			}
-		} else {
-			os.Exit(1)
-		}
+	if err := os.MkdirAll(config.CloneIntoDir, 0700); err != nil {
```

Comment:
please never use `700` unless required, this will make it only accessible for the user, which might conflict with other, usually system/daemon tools

Reply:
in general `755` should be the safe default ; if you do a `git clone` in the terminal, or just a `mkdir -p` then the created dir will be `755`